test-clients/src/main/java/com/hedera/services/bdd/spec/queries/HapiQueryOp.java

```diff
@@ -93,14 +93,21 @@ protected long costFrom(Response response) throws Throwable {
 			if (permissibleCostAnswerPrechecks.get().contains(actualPrecheck)) {
 				costAnswerPrecheck = Optional.of(actualPrecheck);
 			} else {
-				Assert.fail(
+				log.error(
 						String.format(
 								"Cost-answer precheck was %s, not one of %s!",
 								actualPrecheck,
 								permissibleCostAnswerPrechecks.get()));
+
+				throw new Exception(String.format("Cost-answer precheck was %s, not one of %s!",
+						actualPrecheck,
+						permissibleAnswerOnlyPrechecks.get()));
```

Comment:
Assign the `String.format` to a variable to remove duplicate.

Reply:
Done.